src/OAuth/OAuthBuilder.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Penneo\SDK\OAuth;
+
+use GuzzleHttp\Client;
+use Penneo\SDK\OAuth\Config\OAuthConfig;
+use Penneo\SDK\PenneoSDKException;
+
+final class OAuthBuilder
+{
+    /** @var string */
+    private $environment;
+    /** @var string */
+    private $clientId;
+    /** @var string */
+    private $clientSecret;
+    /** @var string */
+    private $redirectUri;
+
+    private function __construct()
+    {
+    }
+
+    public static function start(): self
+    {
+        return new self();
+    }
+
+    public function setEnvironment(string $environment): self
+    {
+        $this->environment = $environment;
+        return $this;
+    }
+
+    public function setClientId(string $clientId): self
+    {
+        $this->clientId = $clientId;
+        return $this;
+    }
+
+    public function setClientSecret(string $secret): self
+    {
+        $this->clientSecret = $secret;
+        return $this;
+    }
+
+    public function setRedirectUri(string $redirectUri): self
+    {
+        $this->redirectUri = $redirectUri;
+        return $this;
+    }
+
+    public function build(Client $client = null): OAuth
+    {
+        $this->validateAllParametersPresent();
+        $this->validateEnvironment();
+        $this->validateRedirectUri();
+
+        return new OAuth(
+            new OAuthConfig(
+                $this->environment,
+                $this->clientId,
+                $this->clientSecret,
+                $this->redirectUri
+            )
+        );
+    }
+
+    /** @throws PenneoSDKException */
+    private function validateAllParametersPresent(): void
+    {
+        if (!$this->environment) {
+            $this->throwMissingParameterError('environment');
+        }
+        if (!$this->clientId) {
+            $this->throwMissingParameterError('clientId');
+        }
+        if (!$this->clientSecret) {
+            $this->throwMissingParameterError('clientSecret');
+        }
+        if (!$this->redirectUri) {
+            $this->throwMissingParameterError('redirectUri');
+        }
+    }
+
+    /** @throws PenneoSDKException */
+    private function throwMissingParameterError(string $missingParameter): void
+    {
+        $capitalized = ucfirst($missingParameter);
+        throw new PenneoSDKException("Cannot build! Please set the {$missingParameter} with ->set{$capitalized}()!");
+    }
+
+    /** @throws PenneoSDKException */
+    private function validateEnvironment(): void
+    {
+        if ($this->environment != 'sandbox' && $this->environment != 'production') {
```

Comment:
Great suggestion. I've implemented it, although I have named it slightly differently, because it reads better -> `Environment::isSupported(string $environment): bool`